modules/apps/commerce/commerce-price-list-service/src/main/java/com/liferay/commerce/price/list/internal/model/listener/CPInstanceUnitOfMeasureModelListener.java

```diff
@@ -0,0 +1,178 @@
+						cpInstance.getCPInstanceUuid(),
+						cpInstanceUnitOfMeasure.getIncrementalOrderQuantity(),
+						cpInstanceUnitOfMeasure.getKey());
+
+				for (CommercePriceEntry commercePriceEntry :
+						commercePriceEntries) {
+
+					commercePriceEntry.setQuantity(BigDecimal.ONE);
+					commercePriceEntry.setUnitOfMeasureKey(null);
+
+					_commercePriceEntryLocalService.updateCommercePriceEntry(
+						commercePriceEntry);
+
+					List<CommerceTierPriceEntry> commerceTierPriceEntries =
+						_commerceTierPriceEntryLocalService.
+							getCommerceTierPriceEntries(
+								commercePriceEntry.getCommercePriceEntryId(),
+								QueryUtil.ALL_POS, QueryUtil.ALL_POS);
+
+					for (CommerceTierPriceEntry commerceTierPriceEntry :
+							commerceTierPriceEntries) {
+
+						commerceTierPriceEntry.setMinQuantity(BigDecimal.ONE);
+
+						_commerceTierPriceEntryLocalService.
+							updateCommerceTierPriceEntry(
+								commerceTierPriceEntry);
+					}
+				}
+			}
+			else {
+				_commercePriceEntryLocalService.deleteCommercePriceEntries(
+					cpInstance.getCPInstanceUuid(),
+					cpInstanceUnitOfMeasure.getIncrementalOrderQuantity(),
+					cpInstanceUnitOfMeasure.getKey());
+			}
+		}
+		catch (PortalException portalException) {
+			if (_log.isWarnEnabled()) {
+				_log.warn(portalException);
+			}
+		}
+	}
+
+	@Override
+	public void onAfterUpdate(
+			CPInstanceUnitOfMeasure originalCPInstanceUnitOfMeasure,
+			CPInstanceUnitOfMeasure cpInstanceUnitOfMeasure)
+		throws ModelListenerException {
+
+		CPInstance cpInstance = _cpInstanceLocalService.fetchCPInstance(
+			cpInstanceUnitOfMeasure.getCPInstanceId());
+
+		List<CommercePriceEntry> commercePriceEntries =
+			_commercePriceEntryLocalService.getCommercePriceEntries(
+				cpInstance.getCPInstanceUuid(),
+				originalCPInstanceUnitOfMeasure.getIncrementalOrderQuantity(),
+				originalCPInstanceUnitOfMeasure.getKey());
+
+		BigDecimal incrementalOrderQuantity =
+			cpInstanceUnitOfMeasure.getIncrementalOrderQuantity();
+		String unitOfMeasureKey = cpInstanceUnitOfMeasure.getKey();
+
+		for (CommercePriceEntry commercePriceEntry : commercePriceEntries) {
+			commercePriceEntry.setQuantity(incrementalOrderQuantity);
+
+			commercePriceEntry.setUnitOfMeasureKey(unitOfMeasureKey);
+
+			_commercePriceEntryLocalService.updateCommercePriceEntry(
+				commercePriceEntry);
+
+			List<CommerceTierPriceEntry> commerceTierPriceEntries =
+				_commerceTierPriceEntryLocalService.getCommerceTierPriceEntries(
+					commercePriceEntry.getCommercePriceEntryId(),
+					QueryUtil.ALL_POS, QueryUtil.ALL_POS);
+
+			for (CommerceTierPriceEntry commerceTierPriceEntry :
+					commerceTierPriceEntries) {
+
+				commerceTierPriceEntry.setMinQuantity(incrementalOrderQuantity);
+
+				_commerceTierPriceEntryLocalService.
+					updateCommerceTierPriceEntry(commerceTierPriceEntry);
+			}
+		}
+	}
+
+	private boolean _isFirstOneCPInstanceUnitOfMeasure(long cpInstanceId)
```

Comment:
don't like this name `_isFirstOneCPInstanceUnitOfMeasure` just remove One and keep `_isFirstCPInstanceUnitOfMeasure`.

Reply:
it's also used in the remove method, in that case it would be the `last` `CPInstanceUnitOfMeasure`